app/src/main/java/com/google/samples/apps/sunflower/PlantDetailFragment.kt

```diff
@@ -60,10 +62,16 @@ class PlantDetailFragment : Fragment() {
             setLifecycleOwner(this@PlantDetailFragment)
             fab.setOnClickListener { view ->
                 plantDetailViewModel.addPlantToGarden()
-                Snackbar.make(view, R.string.added_plant_to_garden, Snackbar.LENGTH_LONG).show()
+                plantDetailViewModel.showSnackbarMessage(R.string.added_plant_to_garden)
```

Comment:
The Line 64-65 should happen asynchronously, that means the `message` would be shown just after `adding` behavior being caused which is incorrect information while the `inserting` is still running underway possibly.

**My opinion**  here is to use the **return** value(`LiveData<Long>`) of insert DAO which can be used by the view-model and show  `message` asynchronously.